src/index.tsx

```diff
@@ -195,19 +190,31 @@ class PyflyByWidget extends Widget {
     }
   }
 
+  async waitForLock() {
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
```

Comment:
Maybe lets use a pattern like this? https://medium.com/@chris_marois/asynchronous-locks-in-modern-javascript-8142c877baf (just be a bit careful as their enable can be called twice and cause some problems).

Reply:
This now uses a set of active locks to execute the requests "sequentially".